src/dlgColorTrigger.cpp

```diff
@@ -23,405 +23,430 @@
-    pushButton_Lyellow->setStyleSheet(styleSheet);
-
-    connect(pushButton_cyan, &QAbstractButton::clicked, this, &dlgColorTrigger::setColorCyan);
-    color = mpTrigger->mpHost->mCyan;
-    palette.setColor(QPalette::Button, color);
-    styleSheet = QStringLiteral("QPushButton{background-color:") + color.name() + QStringLiteral(";}");
-    pushButton_cyan->setStyleSheet(styleSheet);
-
-    connect(pushButton_Lcyan, &QAbstractButton::clicked, this, &dlgColorTrigger::setColorLightCyan);
-    color = mpTrigger->mpHost->mLightCyan;
-    palette.setColor(QPalette::Button, color);
-    styleSheet = QStringLiteral("QPushButton{background-color:") + color.name() + QStringLiteral(";}");
-    pushButton_Lcyan->setStyleSheet(styleSheet);
-
-    connect(pushButton_magenta, &QAbstractButton::clicked, this, &dlgColorTrigger::setColorMagenta);
-    color = mpTrigger->mpHost->mMagenta;
-    palette.setColor(QPalette::Button, color);
-    styleSheet = QStringLiteral("QPushButton{background-color:") + color.name() + QStringLiteral(";}");
-    pushButton_magenta->setStyleSheet(styleSheet);
-
-    connect(pushButton_Lmagenta, &QAbstractButton::clicked, this, &dlgColorTrigger::setColorLightMagenta);
-    color = mpTrigger->mpHost->mLightMagenta;
-    palette.setColor(QPalette::Button, color);
-    styleSheet = QStringLiteral("QPushButton{background-color:") + color.name() + QStringLiteral(";}");
-    pushButton_Lmagenta->setStyleSheet(styleSheet);
-
-    connect(pushButton_white, &QAbstractButton::clicked, this, &dlgColorTrigger::setColorWhite);
-    color = mpTrigger->mpHost->mWhite;
-    palette.setColor(QPalette::Button, color);
-    styleSheet = QStringLiteral("QPushButton{background-color:") + color.name() + QStringLiteral(";}");
-    pushButton_white->setStyleSheet(styleSheet);
-
-    connect(pushButton_Lwhite, &QAbstractButton::clicked, this, &dlgColorTrigger::setColorLightWhite);
-    color = mpTrigger->mpHost->mLightWhite;
-    palette.setColor(QPalette::Button, color);
-    styleSheet = QStringLiteral("QPushButton{background-color:") + color.name() + QStringLiteral(";}");
-    pushButton_Lwhite->setStyleSheet(styleSheet);
-}
 
-void dlgColorTrigger::setColorBlack()
-{
-    if (!mpTrigger) {
-        return;
-    }
-    mpTrigger->mColorTrigger = true;
-    if (mMode == 0) {
-        mpTrigger->mColorTriggerFg = true;
-        mpTrigger->mColorTriggerFgAnsi = 2;
-        mpTrigger->mColorTriggerFgColor = mpTrigger->mpHost->mBlack;
+    mSignalMapper = new QSignalMapper(this);
+
+    // The buttonBox's Cancel button has been connect up to this class's
+    // rejected() signal to close it without changing anything in the caller!
+    // The ignore button means do not consider this fore or back ground (as per
+    // mIsBackground) for the colour trigger - only use the other one (which
+    // must be set)
+    connect(buttonBox->button(QDialogButtonBox::Ignore), SIGNAL(pressed()), this, SLOT(slot_resetColorClicked()));
```

Comment:
Let's use the functor-based connectors here - QtCreator offers a right-click refactor option to update this.

Reply:
Does it *now*? I hadn't seen that but if it is there that makes things simpler...

Will see if I can do that...

Reply:
Ah, it is a feature added in Qt Creator 4.7.0...

... it does not handle code using `QSignalMapper` because Qt claims that is obsolete now that lambda functions can be used as slot methods.  However I cannot quite get my head around that at present.

Reply:
That feature has been around for quite a while in Qt Creator. What do we need `QSignalMapper` for? I don't think that is used in this connect.

Reply:
I used it because it enables a single slot to be used for multiple signals from similar but distinguishable controls - i.e. you set up a mapping of the signal generators then the slot can tell which control sent the signal and can behave accordingly.  In this situation it eliminated 15 of 16 previous slot methods that each handled a *single* colour selection button - combining all the signals into a larger single slot method.